doc/ledger-reward-events.md

```diff
@@ -0,0 +1,47 @@
+# Ledger Reward Events
+
+This is an ambit claim of what data the ledger events should provide from the point of view of
+`db-sync`.
+
+## Prologue
+
+In the context of `db-sync` and the `ledger`, the word "reward" is slightly misleading and should
+not be narrowly defined as only being payments of staking rewards, but defined much more broadly as
+any and all payments to a stake address. None of these payments are recorded on the block chain as
+part of regular transactions.
+
+Payments to stake addresses are made for one of following reasons:
+* Staking rewards for being a member of a stake pool.
+* Staking rewards for being an owner of a stake pool.
+* MIR payments from the treasury.
+* MIR payments from the reserves.
+* Refunds of the stake pool deposit payment when a pool is de-registered.
+
+Currently, the ledger provides the following reward related events:
+* `LEDeltaReward`
+* `LEMirTransfer`
+* `LERetiredPools`
+* `LETotalRewards`
+
+These events are currently only used by `db-sync` but it is my understanding that in the future,
+other programs will also start using these.
+
+
+## The Desired Ledger Event Functionality
+
+* Every event should carry an `EpochNo` field to make debugging and validation on the `db-sync` side
+  easier.
+* A single `LETotalRewards` event should be emitted for every epoch in the Shelley era and later,
+  even if there are no rewards for that epoch (in which case the event reward map will be `mempty`).
+* `LETotalRewards` should include all payments to stake addresses for a given epoch. That means
+  all staking rewards (member and owner), all MIR payments and all stake pool deposit refunds.
```

Comment:
There is currently no such `LETotalRewards` event in the Cardano API, do you meant the `LERewardEvent`? The MIR payments and pool deposit refunds come from totally different ledger rules, we cannot make the event described here.

Reply:
> There is currently no such LETotalRewards event in the Cardano API,

`db-sync` currently does not use `cardano-api` for anything ledger event related.

Is there any accomodation `ledger` might make the use of these events by the current one and only consumer of these events easier?

Reply:
> currently does not use cardano-api for anything ledger event related.

oh! Where does `LETotalRewards` come from? It's not from cardano-ledger.

> Is there any accomodation ledger might make the use of these events by the current one and only consumer of these events easier?

Absolutely! We just can't glue these events together in the way you are proposing, not with serious consequences